### ISSUE TITLE
Cannot create empty HDF5 file using HDF5 1.8.4 (works fine with 1.8.13 and greater)

### ISSUE THREAD

**Issue by ElDeveloper** [h5py/h5py#525]
When creating an empty dataset using HDF5 1.8.4, the following exception is raised:

```python
/usr/local/lib/python2.7/dist-packages/h5py/_hl/group.pyc in create_dataset(self, name, shape, dtype, data, **kwds)
     97         """
     98         with phil:
---> 99             dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
    100             dset = dataset.Dataset(dsid)
    101             if name is not None:

/usr/local/lib/python2.7/dist-packages/h5py/_hl/dataset.pyc in make_new_dset(parent, shape, dtype, data, chunks, compression, shuffle, fletcher32, maxshape, compression_opts, fillvalue, scaleoffset, track_times)
    112     if maxshape is not None:
    113         maxshape = tuple(m if m is not None else h5s.UNLIMITED for m in maxshape)
--> 114     sid = h5s.create_simple(shape, maxshape)
    115
    116

/usr/local/lib/python2.7/dist-packages/h5py/_objects.so in h5py._objects.with_phil.wrapper (/tmp/pip-build-8Ijz6u/h5py/h5py/_objects.c:2405)()

/usr/local/lib/python2.7/dist-packages/h5py/_objects.so in h5py._objects.with_phil.wrapper (/tmp/pip-build-8Ijz6u/h5py/h5py/_objects.c:2362)()

/usr/local/lib/python2.7/dist-packages/h5py/h5s.so in h5py.h5s.create_simple (/tmp/pip-build-8Ijz6u/h5py/h5py/h5s.c:1353)()

ValueError: Zero sized dimension for non-unlimited dimension (Zero sized dimension for non-unlimited dimension)
```

However, if the libraries are updated to something more recent, the problem goes away. At this point, it is unclear if this is really a bug of h5py or a limitation of those libraries. These tests were done using h5py 2.4.0.

**Comment by andrewcollette** [comment]
Yes, this is an annoying limitation of older versions of HDF5 (it had something to do with the value "0" being reserved).  It was eventually fixed in HDF5 1.8.7:

http://hdfgroup.org/ftp/HDF5/prev-releases/ReleaseFiles/release5-187
(under "Library")

There are lots of issues like this with HDF5 1.8.4; unfortunately it's still used with many LTS-style distributions.  My advice would be to upgrade HDF5 if possible.

**Comment by ElDeveloper** [comment]
Thanks for your quick reply and thanks for the link, this is very
useful.
